lib/posix/spawn.rb

```diff
@@ -271,7 +277,7 @@ def system(*args)
     # Returns the String output of the command.
     def `(cmd)
       r, w = IO.pipe
-      pid = spawn(*system_command_prefixes, cmd, :out => w, r => :close)
+      pid = spawn(*(system_command_prefixes << cmd << {:out => w, r => :close}))
```

Comment:
Does `system_command_prefixes` return a new array object each time? May be safer to use `+` instead of `<<` here. Ditto below.